app/controllers/items_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ItemsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_wishlist!, only: %i(new create)
+
+  def new
+    @item = @wishlist.items.new
+    authorize! :new, @item
+  rescue CanCan::AccessDenied
+    redirect_to @wishlist, alert: t("generic_error")
+  end
+
+  def create
+    @item = @wishlist.items.create(item_params)
+    authorize! :create, @item
+    @item.save!
+    redirect_to @wishlist, notice: t(".success")
+  rescue CanCan::AccessDenied
+    redirect_to @wishlist, alert: t("generic_error")
+  rescue ActiveRecord::RecordInvalid
+    render :new
+  end
+
+  private
+
+  def item_params
+    params[:item] ||= {}
+    params.require(:item).permit(:wishlist_id, :name, :description)
+  end
+
+  def set_wishlist!
+    @wishlist = Wishlist.find(params[:wishlist_id])
+  rescue ActiveRecord::RecordNotFound
+    redirect_to wishlists_path, alert: t("generic_error")
+  end
```

Comment:
In WishlistController we rescued only RecordNotFound error in this method - I think it's not necessary, as after finding the specific wishlist, we continue to new action for item and then check for access. But I can be wrong :)